VerticalSlice2D/Assets/Scripts/TPBar.cs

```diff
@@ -0,0 +1,74 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+
+public class TPBar : MonoBehaviour
+{
+    public float tp = 0;
+    private float startingY;
+    [SerializeField] private TextMeshProUGUI tpText;
+    [SerializeField] private GameObject orangeTPBar;
+    [SerializeField] private GameObject max;
+    [SerializeField] private GameObject tpPercentage;
+    [SerializeField] Transform spriteMask;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        
+        startingY = spriteMask.position.y;
+    }
+
+    void UpdateSprite()
+    {
+        //this is very precise and hopefully still works after moving it please dont kill me
+        spriteMask.position = new Vector2(spriteMask.position.x, startingY+(0.02805f*tp));
+        //0.02805f is de sprite hoogte. Zodra wij sprites krijgen MOET DIT AANGEPAST WORDEN OM GOED TE WERKEN. Ik zou er een functie voor maken maar ik gebruik hier geen spriterenderer, dus kan ik niet de grootte aanroepen.
+    }
+
+    void UpdateTextAndTp()
+    {
+
+        if (tp < 0)
+        {
+            tp = 0;
+        }
+        
+        if (tp < 100)
+        {
+            tpText.text = tp.ToString();
+            orangeTPBar.SetActive(true);
+            max.SetActive(false);
+            tpPercentage.SetActive(true);
+        }
+       
+        if (tp >= 100)
+        {
+            tp = 100;
+            tpText.text = "";
+            orangeTPBar.SetActive(false);
+            max.SetActive(true);
+            tpPercentage.SetActive(false);
+        }
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+    }
+
+    public void AddTp(float amount)
+    {
+        tp += amount;
+        UpdateSprite();
+        UpdateTextAndTp();
+    }
+    public void RemoveTp(float amount)
+    {
+        tp -= amount;
+        UpdateSprite();
+        UpdateTextAndTp();
```

Comment:
Misschien moet de volgorde van deze functies andersom, want de tp kan nog aangepast worden nadat de sprite is aangepast.